test/test_mbjson.py

```diff
@@ -149,6 +150,53 @@ def test_media_formats_from_node(self):
         formats = media_formats_from_node(self.json_doc['media'])
         self.assertEqual(formats, '12" Vinyl')
 
+    def test_originaldate_recording(self):
+        config.setting["originaldate_use_recording"] = True
+        metadata = Metadata()
+        release_group_to_metadata(self.json_doc['release-group'], metadata)
+        release_to_metadata(self.json_doc, metadata)
+        self.assertEqual('1973-03-24', metadata['originaldate'])
+        # Track 1 has a custom first release date
+        track1 = self.json_doc['media'][0]['tracks'][0]
+        recording_metadata = Metadata(metadata)
+        recording_to_metadata(track1['recording'], recording_metadata)
+        self.assertEqual('1972-02-23', recording_metadata['originaldate'])
+        self.assertEqual('1972', recording_metadata['originalyear'])
+        # Track 2 has the same first release date set as the release
+        track2 = self.json_doc['media'][0]['tracks'][1]
+        recording_metadata = Metadata(metadata)
+        recording_to_metadata(track2['recording'], recording_metadata)
+        self.assertEqual('1973-03-24', recording_metadata['originaldate'])
+        self.assertEqual('1973', recording_metadata['originalyear'])
+        # Track 3 has no first release date set
+        track2 = self.json_doc['media'][0]['tracks'][2]
+        recording_metadata = Metadata(metadata)
+        recording_to_metadata(track2['recording'], recording_metadata)
+        self.assertEqual('1973-03-24', recording_metadata['originaldate'])
+        self.assertEqual('1973', recording_metadata['originalyear'])
+
```

Comment:
Hard to read, is it possible to split in smaller functions? or, at least, add few blank lines

Reply:
I split the tests into smaller tests, more limited to testing the actual code.